lib/sqlite_ecto/error.ex

```diff
@@ -8,5 +8,15 @@ defmodule Sqlite.Ecto.Error do
   def exception({type, msg}), do: %__MODULE__{sqlite: {type, to_string(msg)}}
   def exception(msg), do: %__MODULE__{sqlite: msg}
 
+  require Logger
+
+  def to_constraints(%__MODULE__{sqlite: {:constraint, "UNIQUE constraint failed: " <> _}}) do
```

Comment:
It is impossible to do it in the adapter. to_constraints is called for all errors, regardless if the user called the constraint functions, because we still show a proper error message.

> If I add the warning to the exception, users will still see the message about being unable to support constraints whether they are using them or not. 

I wouldn't worry about it because it is exactly what ecto does. We will always mention changesets in constraint errors because that's the best way for users to learn, I think. If we are wrong, we can just change in the future.